docs/pages/guides/editing-richtext.mdx

```diff
@@ -0,0 +1,118 @@
+
+# Editing rich text
+
+There are a few possible approches to get rich text rendering to work, but all have different limitations.
+
+## Webview-based editors
+
+While most react-native UI components wrap native platform primitives and are fast, performant and “native feeling” as a result, the webview-based rich text editors use a different approach.
+
+They wrap an existing rich text editor built for web with JavaScript inside a [react-native-webview](https://docs.expo.dev/versions/latest/sdk/webview/). It works on all platforms (Android, iOS, Web) and can take advantage of popular rich text editors available for the Web platform, but has a performance and UX penalty.
+
+You will not be able to use native UI components inside the editor. Any implementation of features like mentions or image embedding will duplicate features and require significant effort to implement.
+
+### Existing webview-based react-native packages
+
+There are a couple of existing react-native packages to allow rich-text editing. These are the easiest option to get started if you just need a basic rich text editor with limited configurability and don’t have strict performance or UX requirements.
+
+Here are some options:
+
+- [react-native-rich-editor](https://github.com/wxik/react-native-rich-editor)
+- [react-native-cn-quill](https://github.com/imnapo/react-native-cn-quill)
+
+### Custom webview-based editor
+
+If you need more configurability, you can build a similar package on your own with an existing web only editor, but handle the message passing and web implementation yourself. This gives you all the options your underlying editor offers you and let’s you implement more features.
+
+Here are some options:
+
+- [Quill](https://quilljs.com/)
+- [lexical](https://github.com/facebook/lexical)
+- [slate](https://github.com/ianstormtaylor/slate)
+
+You will need to use message passing to pass text and `onChange` events to and from the webview. Since rich texts often end up long, it’s better to model it as an uncontrolled component to prevent lag on each keystroke. Also, if you can avoid serializing and sending the entire state on each keystroke that also improves performance.
+
+## Building on top of React Native TextInput
+
+Let’s discuss if a feature-complete, general purpose rich text input is possible.
+
+React Native allows nested `<Text>` tags, and it even allows them to be used as children of `<TextInput>` to render and edit styled text. It even is synchronous on new react native architecture (the onChange event fires immediately after a new character is typed into the text input field).
+
+Unfortunately, the `<TextInput>` component is built towards working with regular text, and it only returns a string in it’s `onTextChange` callback. This is a significant limitation.
+
+Here’s a quick example. If you are rendering a text input with the following content:
+
+**aa**aa
```

Comment:
```suggestion
```

I think it is easy to read the text nesting in the code snippet below, unless I'm missing the significance of this line.